cppcache/include/geode/CacheAttributes.hpp

```diff
@@ -80,17 +82,16 @@ class CPPCACHE_EXPORT CacheAttributes {
   /** Sets cache level endpoints list.
    *
    */
-  void setEndpoints(char* endpoints);
+  void setEndpoints(const std::string& endpoints);
+  void setEndpoints(std::string&& endpoints);
```

Comment:
I think I was experimenting here. I should back it out. We can always add it later.

```void setEndpoints(const std::string& endpoints);```
Does not result in a copy to make the call but since internally we keep it a copy is made. No optimization can be made but none is really necessary unless the param is an r-value. The method also fails to communicate that we are keeping a copy.

```void setEndpoints(std::string endpoints);```
Communicates we are keeping a copy as copy is don't on the call. Unoptimized the copy happens also at the assignment to the member variable, though most modern compilers I believe will optimize that second copy out.

```void setEndpoints(std::string&& endpoints);```
Communicates a copy is kept and optimizes the r-value to avoid extra copies.

The two methods implemented are consistent with what the container class templates generate to handle l-values and r-values optimally.